core/src/Kokkos_NumericTraits.hpp

```diff
@@ -141,9 +142,9 @@ template <> struct digits_helper<double> { static constexpr int value = DBL_MANT
 template <> struct digits_helper<long double> { static constexpr int value = LDBL_MANT_DIG; };
 template <class> struct digits10_helper {};
 template <> struct digits10_helper<bool> { static constexpr int value = 0; };
-constexpr double log10_2 = 2.41;
+
 #define DIGITS10_HELPER_INTEGRAL(TYPE) \
-template <> struct digits10_helper<TYPE> { static constexpr int value = digits_helper<TYPE>::value * log10_2; };
+template <> struct digits10_helper<TYPE> { static constexpr int value = std::numeric_limits<TYPE>::digits10; };
```

Comment:
You can't use `std::numeric_limits`. This is not available in all backends and for all relevant configurations.

Reply:
Ok I've fixed that. Was not aware sorry.